src/main/java/com/example/springbootbookshop/dto/CreateBookRequestDto.java

```diff
@@ -0,0 +1,11 @@
+package com.example.springbootbookshop.dto;
+
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import java.math.BigDecimal;
+
+public record CreateBookRequestDto(@NotNull String title, @NotNull String author,
```

Comment:
Maybe NotBlank will be more suitable for String validation. As NotNull not allow only null values, but field can be empty. As for NotBlank -> a String field constrained with @NotBlank must be not null, and the trimmed length must be greater than zero